.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,124 @@
+on:
+  schedule:
+    - cron: 10 0 * * 1
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    steps:
+      # Set fetch-depth: 0 to fetch commit history and tags for use in version calculation
+      - name: Check out code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == github.repository
+        with:
+          username: ${{ secrets.DOCKERHUB_READ_USER }}
+          password: ${{ secrets.DOCKERHUB_READ_TOKEN }}
+
+      - name: Build with Gradle
+        uses: hypertrace/github-actions/gradle@main
+        with:
+          args: build dockerBuildImages
+
+      - name: Run Trivy vulnerability scanner for entity-service image
+        uses: hypertrace/github-actions/trivy-image-scan@main
+        with:
+          image: hypertrace/entity-service
+          output-mode: github
+
+  test:
+    runs-on: ubuntu-22.04
+    steps:
+      # Set fetch-depth: 0 to fetch commit history and tags for use in version calculation
+      - name: Check out code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Unit test
+        uses: hypertrace/github-actions/gradle@main
+        with: 
+          args: jacocoTestReport
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
+        with:
+          name: unit test reports
+          flags: unit
+
+      - name: Integration test
+        uses: hypertrace/github-actions/gradle@main
+        with: 
+          args: jacocoIntegrationTestReport
+  
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
+        with:
+          name: integration test reports
+          flags: integration
+  
+      - name: copy test reports
+        uses: hypertrace/github-actions/gradle@main
+        with: 
+          args: copyAllReports --output-dir=/tmp/test-reports
+        if: failure()
+            
+      - name: Archive test reports
+        uses: actions/upload-artifact@v3
+        with:
+          name: test-reports
+          path: /tmp/test-reports
+        if: failure()
+     
+      - name: Publish Unit Test Results
+        uses: EnricoMi/publish-unit-test-result-action@v2
+        if: (success() || failure()) && (github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == github.repository)
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          files: ./**/build/test-results/**/*.xml
+
+  dependency-check:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Dependency Check
+        uses: hypertrace/github-actions/dependency-check@main
+
+  validate-protos:
+    runs-on: ubuntu-22.04
+    container:
+      image: bufbuild/buf:0.56.0
+      credentials:
+        username: ${{ secrets.DOCKERHUB_READ_USER }}
```

Comment:
Shouldn't use creds here. This is using the old buf version, we should upgrade to the GA buf (1.0+).

That'll look like 
```yaml
      - name: Setup buf
        uses: bufbuild/buf-setup-action@v1
        with:
          github_token: ${{ github.token }}
      - name: Lint protos
        uses: bufbuild/buf-lint-action@v1
      - name: Check for breaking changes
        uses: bufbuild/buf-breaking-action@v1
        with:
          against: '.git#branch=origin/main'
```

It also means the buf config needs to be updated though so my suggestion for now would be to just remove the creds from here - let it run with anonymous pulls and we can come through and upgrade it after.